#10 🎨 feat :  테스트 코드 추가

HeesuYeo opened the pull request:
[🎨 feat : #10 테스트 코드 추가]
### PR 타입
-[✅] 기능 추가
-[✅] 의존성, 환경 변수, 빌드 관련 코드 업데이트

### PR 생성 날짜
23.08.08

### 반영 브랜치
local/main -> origin/main

## 작업 내용
Junit 의존성추가

소형테스트 코드 작성 
<br><br>

## 참고 사항
완벽하게 작성되지 않았습니다
<br><br>

## 관련 이슈



<br><br>

HeesuYeo committed:
:see_no_evil: docs: gitignore 수정

ci/cd를 위한 gitignore 수정

HeesuYeo committed:
:art: feat : #10 테스트 코드 추가
junit, 테스트코드 작성
 Fixes: #10 Related to: #9

HeesuYeo commented:
@f-lab-troy 멘토님 tdd를 공부하면서 테스트코드를 도입해보았습니다

f-lab-troy commented on ConcertEventManagement/src/main/java/io/concertconnect/concerteventmanagement/ConcertEventManagementApplication.java:
```diff
@@ -101,6 +101,82 @@ HTTP Status Codes(200,404 ...)
 Spring Module (jpa를 쉽고 편하게 사용하기위한 라이브러리 같은)
 JPA를 추상화 한 Repository 인터 페이스 제공
 
+*/
```
이 코멘트는 코드보다 논의드린 README에 추가시키면 좋은 내용으로 보입니다.

f-lab-troy commented on ConcertEventManagement/src/main/java/io/concertconnect/concerteventmanagement/config/Config.java:
```diff
@@ -34,7 +34,6 @@ private <T extends RequestMappingInfoHandlerMapping> void customizeSpringfoxHand
                 mappings.addAll(copy);
             }
 
-            @SuppressWarnings("unchecked")
```
좋습니다. 이 어노테이션은 가급적 사용하지 않는 것이 좋습니다.

f-lab-troy commented on ConcertEventManagement/src/main/java/io/concertconnect/concerteventmanagement/controller/EventController.java:
```diff
@@ -1,86 +1,68 @@
+    @PostMapping("/{id}/reserve")
+    public ResponseEntity<String> reserveEvent(@PathVariable int id) {
+        String result = eventService.reserveEvent(id);
+        if ("예약이 성공적으로 완료되었습니다!".equals(result)) {
```
String으로 result를 비교하고 처리하는 것보다 enum 또는 constant로 처리하는 방법을 추천드립니다.
String은 호출 메서드에서 언제든 바뀔 수 있기 때문에 의도치 않은 coupling문제가 될 수 있습니다.

f-lab-troy commented on ConcertEventManagement/src/main/java/io/concertconnect/concerteventmanagement/service/EventService.java:
```diff
@@ -0,0 +1,106 @@
+        if (event.getCapacity() > 0) {
+            event.setCapacity(event.getCapacity() - 1);
+            eventRepository.save(event);
+            return "예약이 성공적으로 완료되었습니다!";
```
위에서 말씀드린대로 내용이 바뀔 수 있는 string보다 enum type과 같은 형식으로 return값을 설정하시길 추천드립니다.

HeesuYeo replied:
위에 컨트롤러와 서비스 메서드의 문제를 enum UpdateResult 메서드를 만들어서 수정하였습니다 PR에 추가 하겠습니다!

f-lab-troy commented on ConcertEventManagement/src/main/java/io/concertconnect/concerteventmanagement/service/EventService.java:
```diff
@@ -0,0 +1,106 @@
+
+        Event existingEvent = optionalEvent.get();
+
+        if (updatedEvent.getCapacity() != null) {
```
이부분의 null체크 반복 부분을 조금 더 깔끔하게 처리할 수 있는 방법이 있을지 고민해보시면 좋습니다.
updateEvent에서도 set하는 부분 겹치는 로직이 있는지 refactoring할 수 있는 부분이 있는지도 검토해보시면 좋을 것 같습니다.

HeesuYeo replied:
partialUpdateEvent에 NULL체크 if문은 Optional로 수정했고
updateEvent는 BeanUtils안에있는 기능으로 수정하였습니다 pr에 추가 하겠습니다!

f-lab-troy commented on ConcertEventManagement/src/test/java/resources/data.sql:
```diff
@@ -0,0 +1,6 @@
+/*INSERT INTO event (id, name, venue, ticketstart_Time, ticketend_Time, start_Time, end_Time)
```
git history가 있으니 코멘트된 부분은 지워주셔도 괜찮습니다.

HeesuYeo committed:
:recycle:refactor : Controller, Service의 코드 리팩토링

Controller => UpdateResult 메스드 추가하여 리턴값 리팩토링

Service =>
partialUpdateEvent메서드 널체크 부분 Optional리팩토링,
updateEvent set부분
BeanUtils메서드 사용

Dockerfile 추가

Fixes: #9

f-lab-troy reviewed (approved):
수고하셨습니다

HeesuYeo committed:
Update README.md

HeesuYeo committed:
Merge branch 'f-lab-edu:main' into main

HeesuYeo committed:
Update README.md
